Rubberduck.CodeAnalysis/QuickFixes/IntroduceLocalVariableQuickFix.cs

```diff
@@ -26,23 +27,58 @@ public override void Fix(IInspectionResult result)
         {
             var identifierContext = result.Target.Context;
             var enclosingStatmentContext = identifierContext.GetAncestor<VBAParser.BlockStmtContext>();
-            var instruction = IdentifierDeclarationText(result.Target.IdentifierName, EndOfStatementText(enclosingStatmentContext));
+            var instruction = IdentifierDeclarationText(result.Target.IdentifierName, EndOfStatementText(enclosingStatmentContext), FrontPadding(enclosingStatmentContext));
             _state.GetRewriter(result.Target).InsertBefore(enclosingStatmentContext.Start.TokenIndex, instruction);
         }
 
         private string EndOfStatementText(VBAParser.BlockStmtContext context)
         {
-            if (context.TryGetPrecedingContext<VBAParser.EndOfStatementContext>(out var endOfStmtContext))
+            if (!context.TryGetPrecedingContext<VBAParser.IndividualNonEOFEndOfStatementContext>(out var individualEndOfStmtContext))
             {
-                return endOfStmtContext.GetText();
+                return Environment.NewLine;
             }
 
-            return Environment.NewLine;
+            var endOfLine = individualEndOfStmtContext.endOfLine();
+
+            if (endOfLine?.commentOrAnnotation() == null)
+            {
+                return individualEndOfStmtContext.GetText();
+            }
+
+            //There is a comment inside the preceding endOfLine, which we do not want to duplicate.
+            var whitespaceContext = individualEndOfStmtContext.whiteSpace(0);
+            return Environment.NewLine + (whitespaceContext?.GetText() ?? string.Empty);
+        }
+
+        private string FrontPadding(VBAParser.BlockStmtContext context)
+        {
+            var statementLabelContext = context.statementLabelDefinition();
+            if (statementLabelContext == null)
+            {
+                return string.Empty;
+            }
+
+            var statementLabelTextAsWhitespace = ReplaceNonWhitespaceWithSpace(statementLabelContext.GetText());
+            var whitespaceContext = context.whiteSpace();
+            return statementLabelTextAsWhitespace + (whitespaceContext?.GetText() ?? string.Empty);
+        }
+
+        private string ReplaceNonWhitespaceWithSpace(string input)
+        {
+            if (input == null || input.Equals(string.Empty))
+            {
+                return string.Empty;
+            }
+
+            var pattern = @"[^\r\n\t ]";
```

Comment:
Why not `\S`?

Reply:
Is `\s` exactly equivalent to that combination?

Reply:
`\s` is equivalent to `[\f\n\r\t\v\x85\p{Z}]`, soo ... no